executor/executor_xapi.go

```diff
@@ -324,7 +337,7 @@ func (e *XSelectIndexExec) doTableRequest(handles []int64) (*xapi.SelectResult,
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	// TODO: add offset and limit.
+	// the handles are not in original index order, so we can't push limit here.
```

Comment:
If the order matters, can we sort the handles according to index order? Than we can  push limit down.